build/Band_Representations.jl

```diff
@@ -0,0 +1,260 @@
+
+    irvec2d=return_irvec2d(sgnum, D, maximal, timereversal)
+    klabs=Array{String, 1}(undef, 0)
+    kvs=Array{KVec{D}, 1}(undef, 0)
+    for (klab, kv) in klab2kv
+        push!(klabs, klab)
+        push!(kvs, kv)
+    end
+
+    spinful, allpaths, timeinvar=false, false, true
+    
+    lgirreps= maximal ? get_maximal_lgirreps(sgnum, D, timereversal) : get_lgirreps(sgnum, D, timereversal)
+
+    irlabs=[([[lgg.cdml for lgg in lg] for (klab, lg) in lgirreps]...)...]
+    irdims=[([[size(lgg.matrices[1])[1] for lgg in lg] for (klab, lg) in lgirreps]...)...]
+
+    wps=get_wycks(sgnum, D)
+    sg=spacegroup(sgnum, D)
+
+    wps_special=String[]
+    sitesyms=String[]
+    bandreps_array=Array{BandRep, 1}(undef, 0)
+
+    vec_index=1
+
+    first_k=irlabs[1][1]
+
+    dim_array=[i[1]==first_k for i in irlabs]
+
+    dim_array=dim_array.*irdims
+    
+    sitegroups=SiteGroup.(Ref(sg), wps)
+
+    maximal_wyckoffs=Array{WyckPos{2}, 1}(undef, 0)
+
+    for maximal_sitegroup in findmaximal(sitegroups)
+        push!(maximal_wyckoffs, maximal_sitegroup.wp)
+    end
+
+    for wp in maximal_wyckoffs
+        sitegroup=SiteGroup(spacegroup(sgnum, D), wp)
+
+        sitesym=_find_isomorphic_parent_pointgroup(sitegroup).label
+
+            sitegroup_symmetry_irreps=site_symmetry_irreps(sitegroup, timereversal)
+            ρ_max=size(sitegroup_symmetry_irreps)[1]
+            for ρ in 1:ρ_max
+                mulliken_label=mulliken(sitegroup_symmetry_irreps[ρ])
+                dim=dot(irvec2d[:, vec_index], dim_array)
+                 band_rep=BandRep(string( wp.mult, wp.letter), sitesym, string( wp.mult, wp.letter) , dim, false, false, irvec2d[:, vec_index], irlabs)
+                 push!(wps_special, string( wp.mult))
+                 push!(sitesyms, sitesym)
+                 push!(bandreps_array, band_rep )
+                 vec_index+=1
+        end
+    end
+    return BandRepSet(sgnum, bandreps_array, kvs, klabs, irlabs, allpaths, spinful, timeinvar)
+end
+
+BRS_VEC_ELEMENTARY_MAXIMAL=BandRepSet[];
+BRS_VEC_ELEMENTARY_ALL=BandRepSet[];
+
+BRS_VEC_ELEMENTARYTR_MAXIMAL=BandRepSet[];
+BRS_VEC_ELEMENTARYTR_ALL=BandRepSet[];
+
+#Below, we make an array of bandrepsets in 2 dimensions for maximal k points
+for br_num in 1:17
+    push!(BRS_VEC_ELEMENTARY_MAXIMAL, make_bandrep_set(br_num, D=2, maximal=true, timereversal=false))
+end
+
+#Below, we make an array of bandrepsets in 2 dimensions for all k points
+for br_num in 1:17
+    push!(BRS_VEC_ELEMENTARY_ALL, make_bandrep_set(br_num, D=2, maximal=false, timereversal=false))
+end
+
+#Below, we make an array of bandrepsets in 2 dimensions for maximal k points
+for br_num in 1:17
+    push!(BRS_VEC_ELEMENTARYTR_MAXIMAL, make_bandrep_set(br_num, D=2, maximal=true, timereversal=true))
+end
+
+#Below, we make an array of bandrepsets in 2 dimensions for all k points
+for br_num in 1:17
+    push!(BRS_VEC_ELEMENTARYTR_ALL, make_bandrep_set(br_num, D=2, maximal=false, timereversal=true))
+end
```

Comment:
Pretty sure I wrote this twice already, but this can just be:
```suggestion
# maximal k-points & without time-reversal
BRS_VEC_ELEMENTARY_MAXIMAL   = make_bandrep_set.(1:17, D=2, maximal=true,  timereversal=false)
# all k-points & without time-reversal
BRS_VEC_ELEMENTARY_ALL       = make_bandrep_set.(1:17, D=2, maximal=false, timereversal=false)
# maximal k-points & with time-reversal
BRS_VEC_ELEMENTARYTR_MAXIMAL = make_bandrep_set.(1:17, D=2, maximal=true,  timereversal=true)
# all k-points & with time-reversal
BRS_VEC_ELEMENTARYTR_ALL     = make_bandrep_set.(1:17, D=2, maximal=false, timereversal=true)
```